File: frontend/src/microservices/tour-group/objects/TourGroupMessageResponse.ts
import type { TourGroupMessageAttachmentResponse } from './TourGroupMessageAttachmentResponse'
import type { TourGroupMessageReactionResponse } from './TourGroupMessageReactionResponse'

export type TourGroupMessageResponse = {
  messageId: string
  conversationId: string
  messageType: string
  senderUserId: string
  senderDisplayName: string
  senderAvatarUrl: string | null
  content: string
  replyToMessageId: string | null
  replyToPreview: string | null
  status: string
  createdAt: string
  updatedAt: string
  attachments: TourGroupMessageAttachmentResponse[]
  reactions: TourGroupMessageReactionResponse[]
  canEdit: boolean
  canDelete: boolean
  canRecall: boolean
  canReact: boolean
  isMine: boolean
}
export const tourGroupMessageResponseFromJson = (json: string): TourGroupMessageResponse =>
  JSON.parse(json) as TourGroupMessageResponse

export const tourGroupMessageResponseToJson = (value: TourGroupMessageResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/TourGroupMessageResponse.scala
// TourGroupChatPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import com.typesafe.travel.shared.kernel.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

import java.time.Instant

final case class TourGroupMessageResponse(
    messageId: String,
    conversationId: String,
    messageType: String,
    senderUserId: String,
    senderDisplayName: String,
    senderAvatarUrl: Option[String],
    content: String,
    replyToMessageId: Option[String],
    replyToPreview: Option[String],
    status: String,
    createdAt: String,
    updatedAt: String,
    attachments: List[TourGroupMessageAttachmentResponse],
    reactions: List[TourGroupMessageReactionResponse],
    canEdit: Boolean,
    canDelete: Boolean,
    canRecall: Boolean,
    canReact: Boolean,
    isMine: Boolean
)
object TourGroupMessageResponse:
  given sourceEncoder: Encoder[TourGroupMessageResponse] = deriveEncoder
  given sourceDecoder: Decoder[TourGroupMessageResponse] = deriveDecoder
